### DISCUSSION TITLE
Updates?

### DISCUSSION THREAD

**Discussion by tasi333** [hassio-addons/app-tailscale#113] [Q&A] [answered]
Will this be an automatically updatable add-on, or do we have to do something manually on the command line to get the latest version of tailscale on HA?

I'm currently on 1.26.0, I think the latest version is either 1.30.1 or 1.30.2.

Thanks for any help.

**Comment by frenck** [comment, marked as answer]
Updates are offered via the Home Assistant eco system as per anything in Home Assistant.
There is no update published beyond the point you are at right now.

See also: <https://github.com/hassio-addons/addon-tailscale/releases>

**Reply by tasi333** [reply]
Thanks for the quick response @frenck, what are the plans for the next release?

**Reply by frenck** [reply]
It's on a list, but so are many other things. There is no schedule or plan, this is not a company, but a personal ran open source project.

**Reply by tasi333** [reply]
Understood, I appreciate all your work, and getting this onto HA.  Thank You.

**Reply by bohtho** [reply]
Great to see the add-on updated up to 1.30.1 I notice per the main branch (still unreleased).

So i´m definitely not trying to be an ass here, but maybe throw in 1.30.2 in the addon´s next release too ? I'm not suggesting that the project should track depending software releases by the hour, but it has an interesting changelog with regard to how we run tailscale in a container too. Maybe the outstanding issues here regarding routes etc. will get resolved : https://github.com/tailscale/tailscale/releases/tag/v1.30.2

Anyway, thanks

**Comment by FrankBakermans** [comment]
Hi Frenck,

I appreciate the work you have put in to develop this add on and make Tailscale available in HA. I love Tailscale and I have quite a few systems in my Tailscale network, but I do like to keep the version of Tailscale up to date on all my devices. I see from your answers, that the only way to update the Tailscale software is when a new version of the add-on is released.

Would it be possible to design and build this add-on in such a way, that we (as users) can update the Tailscale software version independently from the add-on version? This would save you work in the long run and us users can have up to date Tailscale software on our HA.

Thanks,
Frank

**Reply by frenck** [reply]
> Would it be possible to design and build this add-on in such a way, that we (as users) can update the Tailscale software version independently from the add-on version?

Sure, but not doing that. If you are looking for that, the use of add-ons and the installation method you are using might not be a fit.
Maybe you should looking rolling plain Docker containers instead of using HAOS in that case.

> This would save you work in the long run 

Incorrect, this would mean that if upstream incompatible changes happen, stuff just magically break. I want people to be able to update with confidence and a piece of mind, not fearing if stuff would still work.

**Reply by tasi333** [reply]
Would just like to add to this conversation, I use PFSense for my router/firewall, and users have so far been successfully able to update the tailscale version independent of the plugin/addon, take a look at this thread, you can see users posting the update package command and URL every time an update is released.

https://forum.netgate.com/topic/174525/how-to-update-to-the-latest-tailscale-version/83

It would be great to have this as an "option" in HA for the more advanced user and keep updates tied to the add-on for the common user.

Just a thought.

**Reply by frenck** [reply]
See my convo above. I will not add this to my repos for the above reasoning. There is no reasoning or discussion that will change that.


It defeats the whole purpose of containers and add-ons. I strongly recommend picking an setup or ecosystem the better fits your needs in that case, as all suggestions in here are not fitting the methodology.

**Reply by lmagyar** [reply]
In each year there is a TS release when we have to modify something in the add-on to make it work.

Personally I value TS add-on's stability infinite times more then to be on the latest: if TS add-on breaks, there is no acces to the device if it is on a CG-NAT-ed ISP.
